download_action.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+)
+
+type DownloadAction struct {
+	BaseAction `yaml:",inline"`
+	Url        string // URL for downloading
+	Filename   string // File name, overrides the name from URL.
+}
+
+// Function for downloading single file object with http(s) protocol
+func DownloadHttpUrl(url, filename string) error {
+	log.Printf("Download started: '%s' -> '%s'\n", url, filename)
+
+	// TODO: Proxy support?
+
+	// Check if file object already exists.
+	// Ignore check if flag 'forcedRewrite' is set
+	fi, err := os.Stat(filename)
+	if !os.IsNotExist(err) && !fi.Mode().IsRegular() {
+		return fmt.Errorf("Failed to download '%s': '%s' is not a regular file\n", url, filename)
```

Comment:
That error should indicate that filename already exists and isn't a regular file (otherwise it's quite unclear what happened)

Reply:
done